File: Tree/LearnYard/Binary_Tree_Inorder_Traversal/Inorder.java
package Tree.LearnYard.Binary_Tree_Inorder_Traversal;

import java.util.ArrayList;

class Node {
    int val;
    Node left, right;
    Node(int val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

public class Inorder {
    public static void inorder_traversal(Node root, ArrayList<Integer> result) {
        if (root == null) {
            return;
        }
        inorder_traversal(root.left, result);
        result.add(root.val);
        inorder_traversal(root.right, result);
    }
    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);
        root.left.left = new Node(4);
        root.left.right = new Node(5);

        ArrayList<Integer> result = new ArrayList<>();
        inorder_traversal(root, result);

        for (int num : result) {
            System.out.print(num + " ");
        }
    }
}

File: Tree/LearnYard/Binary_Tree_Inorder_Traversal/inorder.py
class Node:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def inorder(root, result):
    if root is None:
        return
    inorder(root.left,result)
    result.append(root.val)
    inorder(root.right,result)

# Creating a sample binary tree
#       1
#      / \
#     2   3
#    / \
#   4   5

root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)

result = []
inorder(root, result)
print(result)
